pkgs/by-name/x2/x2t/package.nix

```diff
@@ -0,0 +1,273 @@
+{ 
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  libxml2,
+  boost,
+  gumbo,
+  icu,
+  freetype,
+  qt5,
+  harfbuzz,
+  # needs to be static and built with MD2 support!
+  openssl,
+  # should find out if newer versions work...
+  v8,
+}:
+
+let
+  # see core/Common/3dParty/fetch.sh
+  katana-parser-src = fetchFromGitHub {
+    owner = "jasenhuang";
+    repo = "katana-parser";
+    rev = "be6df458d4540eee375c513958dcb862a391cdd1";
+    hash = "sha256-SYJFLtrg8raGyr3zQIEzZDjHDmMmt+K0po3viipZW5c=";
+  };
+  # see build_tools scripts/core_common/modules/hyphen.py
+  hyphen-src = fetchFromGitHub {
+    owner = "hunspell";
+    repo = "hyphen";
+    rev = "73dd2967c8e1e4f6d7334ee9e539a323d6e66cbd";
+    hash = "sha256-WIHpSkOwHkhMvEKxOlgf6gsPs9T3xkzguD8ONXARf1U=";
+  };
+  # see core/Common/3dParty/html/fetch.py
+  gumbo-parser-src = fetchFromGitHub {
+    owner = "google";
+    repo = "gumbo-parser";
+    rev = "aa91b27b02c0c80c482e24348a457ed7c3c088e0";
+    hash = "sha256-+607iXJxeWKoCwb490pp3mqRZ1fWzxec0tJOEFeHoCs=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "x2t";
+  version = "8.1.1";
+
+  src = /home/aengelen/dev/documentserver;
+  #src = fetchFromGitHub {
+  #  owner = "ONLYOFFICE";
+  #  repo = "DocumentServer";
+  #  rev = "v${finalAttrs.version}";
+  #  fetchSubmodules = true;
+  #  hash = "sha256-GuTDN7KLuhxW4ZHBazbW4Czi0EJoGraZd35MA7g2ElA=";
+  #};
+  nativeBuildInputs = [ pkg-config qt5.full ];
+  buildInputs = [ gumbo boost libxml2.dev icu freetype.dev ];
+  dontStrip = true;
+  buildPhase = ''
+    runHook preBuild
+
+    # https://github.com/onlyOFFICE/build_tools makes many assumptions,
+    # so we do things 'manually' here...
+
+    cd core/Common/3dParty/cryptopp/project
+    qmake -o Makefile cryptopp.pro
+    make
+    cd ../../../../..
+
+    cd core/OOXML/Projects/Linux/XlsbFormatLib
+    qmake -o Makefile XlsbFormatLib.pro
+    make
+    cd ../../../../..
+
+    cd core/MsBinaryFile/Projects/XlsFormatLib/Linux
+    qmake -o Makefile XlsFormatLib.pro
+    make
+    cd ../../../../..
+
+    cd core/OdfFile/Projects/Linux
+    qmake -o Makefile OdfFormatLib.pro
+    make
+    cd ../../../..
+
+    cd core/MsBinaryFile/Projects/DocFormatLib/Linux
+    qmake -o Makefile DocFormatLib.pro
+    make
+    cd ../../../../..
+
+    cd core/MsBinaryFile/Projects/PPTFormatLib/Linux
+    qmake -o Makefile PPTFormatLib.pro
+    make
+    cd ../../../../..
```

Comment:
Can we generate this with nix code?

Reply:
maybe, but I'm not sure the improved brevity would weigh up to the increased complexity.